src/spatialdata/_core/query/relational_query.py

```diff
@@ -129,6 +147,255 @@ def _filter_table_by_elements(
     return table
 
 
+def _create_element_dict(
```

Comment:
I think this function could be replaced by default dicts

Reply:
if creating a get method in `SpatialData` we could clean the code even more and not having to use default dicts.

Reply:
or allow for `"..." in sdata`

Reply:
true, please feel free to go ahead with that implementation in a tiny PR.

Reply:
In sdata is not allowed but did not use it here as we still require the element_type. I used defaultdict now as suggested.